### PR TITLE
LIB 154: Add bulk create

### PR DESCRIPTION
**Pull request by opalczynski**

### CODE REVIEW

File: syncano/models/archetypes.py [hunk 1]
```diff
@@ -104,6 +110,29 @@ def __eq__(self, other):
             return self.pk == other.pk
         return NotImplemented
 
+    @classmethod
+    @contextmanager
+    def bulk(cls, instance):
```
Comment (23doors) [inline syncano/models/archetypes.py]: Instance param shouldn't be required and it should default to instance set in connection

Comment (opalczynski) [inline syncano/models/archetypes.py]: Good point :+1:

**Comment by 23doors** [comment]
While this is an interesting approach and idea of having one context manager to rule them all is very tempting, I imagine this is also wildly confusing to end-user. It is because suddenly, all object operations are kind of asynchronous. Well, "kind of" as it doesn't seem like they are eventually populated anyway, but even if they were - it feels strange. 

About the naming - on backend side - it is batching. Bulk makes sense for bulk_create and not as a general use context manager. Although it apparently didn't make it to this task's description, we initially agreed to make it like Django one:
https://docs.djangoproject.com/en/1.8/ref/models/querysets/#bulk-create

Now other ops will still be a challenge as ideally we could just have:
`Object.batch(Object.please.update(), Object.please.delete())` etc as we would need to make it lazy and we don't want it to be lazy outside of batch.
And we also cannot have:
`Object.batch(Object.please.update, Object.please.delete)` etc as delete and update params can be a part of filter/update

So I guess the easiest way out is to have:
`connection.batch(Object.please.as_batch().update(), etc)` or `connection.batch(Object.please(batch=True).update(), etc)` or `please_batch`/ `batch` as a manager or whatever.
+
`Object.bulk_create(...)` (as a helper for connection.batch most likely)
`bulk_create` would then return a list of created Objects and `batch` - a list of responses/or whatever that call returns.

Which is also much simpler and easier to both use and understand than what you worked on here.

@dancio @Zhebr Thoughts?

**Comment by opalczynski** [comment]
@23doors Looks like description matters ;) I can do it in djago way if that was the decision :)

**Comment by 23doors** [comment]
Ok. Django way + above it is then. ;)

**Comment by mkucharz** [comment]
This is an example why we should spend more time on planning on CORE/LIBRARIES product meeting - let's have it in mind. I hope that soon we will have backlogs full of well documented tasks ;)

**Pull request closed without merging**